app/controllers/pieces_controller.rb

```diff
@@ -1,11 +1,14 @@
 class PiecesController < ApplicationController
-  def show
-    @piece = Piece.find(params[:id])
-  end
-
   def update
-    current_piece.update_attributes(pieces_params)
-    redirect_to game_path(current_piece.game)
+    row = pieces_params[:row].to_i
+    column = pieces_params[:column].to_i
+
+    if current_piece.valid_move?(row, column)
+      current_piece.update_attributes(pieces_params)
+      render text: 'updated!'
+    else
+      render text: 'invalid', status: :bad_request
```

Comment:
@mscoutermarsh Wasn't sure what to do exactly for the render stuff. I have it this way so that the ajax request in the javascript will actually hit the .done() function. Is there another way to only return the http status and still trigger the ajax.done() function?